protocol/chainlib/chain_fetcher.go

```diff
@@ -116,6 +118,23 @@ func (cf *ChainFetcher) Verify(ctx context.Context, verification VerificationCon
 		}...)
 		return nil
 	}
+
+	// craft data for GET_BLOCK_BY_NUM verification that cannot use "earliest"
+	// also check for %d because the data constructed assumes its presence
+	if verification.ParseDirective.FunctionTag == spectypes.FUNCTION_TAG_GET_BLOCK_BY_NUM {
+		if verification.LatestDistance != 0 && latestBlock != 0 {
+			if latestBlock >= verification.LatestDistance {
+				data = []byte(fmt.Sprintf(parsing.FunctionTemplate, latestBlock-verification.LatestDistance))
```

Comment:
we are not checking for %d

Reply:
FUNCTION_TAG_GET_BLOCK_BY_NUM will always have %d, this check is never done in other places in the provider.
should i still add it?

Reply:
yes, add it in the consensus, then you don't need it here

Reply:
done